cro-clib/chain.h

```diff
@@ -0,0 +1,44 @@
+/*
+chain c bindings
+*/
+
+#define MAX_LENGTH 1024
+#define MAX_STRING_LENGTH 512
+/* general byte buffer */
+typedef struct {
+    unsigned char buf[MAX_STRING_LENGTH];
+    int length;
+} Buf;
+
+/* general api result */
+typedef struct {
+    int error;
```

Comment:
I suggest following Cardano's style (https://github.com/input-output-hk/rust-cardano/blob/master/cardano-c/cardano.h#L183) to handle the error.

Reply:
ok, i'll check, 
but using heap is difficult to use.

Reply:
I suggested that yesterday